processor/utils.ts

```diff
@@ -26,6 +26,35 @@ export const getAttrs = <T>(jsx: MdxJsxFlowElement | MdxJsxTextElement) =>
     return memo;
   }, {} as T);
 
+export const getChildren = <T>(jsx: MdxJsxFlowElement | MdxJsxTextElement) =>
+    jsx.children.reduce((memo, child: MdxFlowExpression, i) => {
+     memo[i] = {
+      // TODO: infer type
+      type: 'text',
+      value: child.value,
+      position: child.position,
+     };
+      return memo;
+    }, [] as T);
+
 export const isMDXElement = (node: Node): node is MdxJsxFlowElement | MdxJsxTextElement => {
   return ['mdxJsxFlowElement', 'mdxJsxTextElement'].includes(node.type);
 }
+
+export const formatHTML = (html: string): string => {
+  if (html.startsWith('`') && html.endsWith('`')) {
+    html = html.slice(1, -1);
+  }
+  const cleaned = html.replace(/^\s*\n|\n\s*$/g, '');
+  const tab = cleaned.match(/^\s*/)[0].length;
+  const tabRegex = new RegExp(`^\\s{${tab}}`, 'gm');
+  const unindented = cleaned.replace(tabRegex, '');
+  return unindented;
+}
+
+export const reformatHTML = (html: string, indent = 2) => {
+  const cleaned = html.replace(/^\s*\n|\n\s*$/g, '');
+  const tab = ' '.repeat(indent);
+  const indented = cleaned.split('\n').map((line: string) => `${tab}${line}`).join('\n');
+  return indented;
+}
```

Comment:
These types of manual regex replacers always feel a bit dangerous to me! And on top of that, they're very hard to read at a glance. Can we add some comments clarifying what exactly they're finding/replacing?

Reply:
Could we use `rehype-stringify` for formatting? https://github.com/rehypejs/rehype/tree/main/packages/rehype-stringify

Reply:
@rafegoldberg i was looking at this last night after i put up the PR and was also thinking it definitely needed some comments. 😆 i didn't want to add a whole new 3rd party library just to remove extra tabs/newlines, but keeping the HTML formatted the same as what the user put in (or a close approximation) seemed important. (i was also thinking of capturing how many spaces the user had in an indent/tab and keeping track of it, but that's probably more appropriate as a fast-follow haha)

Reply:
@kellyjosephprice nice! i'll check it out

Reply:
@kellyjosephprice it looks like `rehype-stringify` uses `hast-util-to-html` under the hood, and i'm just trying to pad/unpad the template literal that the HTML comes in, haha. which reminds me, i needed to adjust this in case the user puts their HTML at the same indent as the component itself 👍🏻

Reply:
Yea, it'd probably be like `unified().use(rehypeParse).use(rehypeStringify).runSync(string)` or something. But I see, you also adding indent. I _think_ there's utils for that in `remark-stringify`.

Reply:
https://github.com/readmeio/markdown/blob/023c2e72825de2c1a64a2d044f421215655bd963/processor/compile/callout.ts#L15

Reply:
I think it gets dangerous when you're nested in another container block, ie:
```
> 📘 How to write plain HTML
> 
> <HTMLBlock>{`
>   <span>...</span>
> `}</HTMLBlock>

```

Reply:
the `HTMLBlock` can't (or shouldn't) be nested, though, right? also, the `HTMLBlock` renders the HTML inside the component, not display it as plain text? (unless you had `safeMode` on i think, heh)

Reply:
Anything can be nested, whether or not it should. Like in a blockquote, callout, or list.

Yea, I guess I'm not sure what the actual markdown would be, would my exmaple above actually be:

```
> 📘 How to write plain HTML
> 
> <HTMLBlock>{`
  <span>...</span>
`}</HTMLBlock>
```

I have no idea what the jsx parser/compiler would actually do here.

Reply:
I think the first example is the right one (https://mdxjs.com/playground/):
![image](https://github.com/readmeio/markdown/assets/451488/6ce4af47-7d41-4864-86f6-f255eec84457)

Reply:
hahaha well this just introduced a lot of terrifying scenarios 😂